charts/alertmanager/templates/statefulset.yaml

```diff
@@ -43,10 +43,29 @@ spec:
       nodeSelector:
         {{- toYaml . | nindent 8 }}
       {{- end }}
-    {{- with .Values.affinity }}
+    {{- if or .Values.podAntiAffinity .Values.affinity }}
       affinity:
+    {{- end }}
+    {{- if .Values.affinity }}
         {{- toYaml . | nindent 8 }}
     {{- end }}
```

Comment:
@gitgaoxiang @monotek The `{{- with .Values.affinity }}` is dropped but `{{- toYaml . | nindent 8 }}` is still used. I think the `with` functionality should be used or it should be `{{- toYaml .Values.affinity | nindent 8 }}`